documentation/wiki/decisions/adr/2024-10-18-document-storage.md

```diff
@@ -0,0 +1,93 @@
+- Ease of processing - Use the best tools, with already supported libraries as they're intended
+- Follow best practices
+
+## Options Considered
+
+- [AWS Simple Storage Service (S3)](#aws-simple-storage-service-s3-buckets) bucket(s)
+- [Store files in PostgreSQL](#store-files-in-postgresql)
+- [Other off-the-shelf or homegrown storage solution](#other-off-the-shelf-or-homegrown-storage-solution)
+
+## Decision Outcome <!-- REQUIRED -->
+
+Chosen option: "AWS S3", because it represents the lowest Total Cost of Ownership (TCO) and industry best practices, including baked in support for access control for files, backups, etc. We will use 2 buckets, one for Published Opportunities, and one for DRAFT Opportunities. When an Opportunity is Published we'll ensure that the associated documents are copied to the Published bucket, making them accessible to the general public. Prior to publishing the documents will be accounted for in S3, so that the file storage is consistent throughout the lifecycle, but only the Publishing Service will have access to the files, ensuring they are not released to the public before the Opportunity and they can be revoked from public view if the Opportunity is accidentally Published.
+
+### Positive Consequences <!-- OPTIONAL -->
+
+- Cost can be managed in trade off with performance profile of requests for files
+- Directly integrates with the AWS Content Delivery Network(CDN), CloudFront
+- Existing tooling/API allows for manipulation of files from the Publishing System and manually if needed.
+- S3 API is standard mimicked/supported by other cloud storage providers if we ever wanted to move these files elsewhere.
+
+### Negative Consequences <!-- OPTIONAL -->
+
+- Requires it's own management if we wanted to sync the files to another environment
+- Disconnects the lifecycle from data in the DB, so any archiving/deleting of files doesn't happen automatically
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+### AWS Simple Storage Service (S3) bucket(s)
+
+Utilize the AWS S3 Service to store/host files. This problem is precisely what S3 was built to solve. It provides strong tooling, monitoring, logging, all built and ready to use. We can architect in such a way that files get scanned before being placed in the final bucket, and get very fine grained support for file versioning, backups, lifecycle, etc. <!-- OPTIONAL -->
+
+- **Pros**
+  - Cost can be managed in trade off with performance
+  - Integrates with AWS Content Delivery Network(CDN), CloudFront
+  - Existing tooling/API
+  - S3 API is standard mimicked/supported by other cloud storage providers if we ever wanted to move these files elsewhere.
+  - Built in support for auto-expiring links (which we want at least in the near term until we come up with more of a final structure/naming strategy)
+- **Cons**
+  - Another separate resource to manage if we're trying to sync/simulate Prod with other environments
```

Comment:
We already have an `api-prod-general-purpose` bucket

https://us-east-1.console.aws.amazon.com/s3/buckets?region=us-east-1&bucketType=general&region=us-east-1

Reply:
This bucket is good for the draft opportunities, we'd make a new one for published